src/target.rs

```diff
@@ -0,0 +1,415 @@
+            os: "invalid-none-set".into(),
+            env: "invalid-none-set".into(),
+            abi: "invalid-none-set".into(),
+        };
+
+        for cfg in cfgs.lines() {
+            if let Some((name, value)) = cfg.split_once('=') {
+                // Remove whitespace, if `rustc` decided to insert any
+                let name = name.trim();
+                let value = value.trim();
+
+                // Remove quotes around value
+                let value = value.strip_prefix('"').unwrap_or(value);
+                let value = value.strip_suffix('"').unwrap_or(value);
+
+                let value = value.to_string();
+                match name {
+                    "target_arch" => target.arch = value,
+                    "target_vendor" => target.vendor = value,
+                    "target_os" => target.os = value,
+                    "target_env" => target.env = value,
+                    "target_abi" => target.abi = value,
+                    _ => {}
+                }
+            } else {
+                // Skip cfgs like `debug_assertions` and `unix`.
+            }
+        }
+
+        target
+    }
+
+    #[test]
+    fn parse_rustc_targets() {
+        let target_list = Command::new("rustc")
+            .arg("--print=target-list")
+            .output()
+            .unwrap()
+            .stdout;
+        let target_list = String::from_utf8(target_list).unwrap();
+
+        let mut has_failure = false;
+        for target in target_list.lines() {
+            let cfgs = Command::new("rustc")
+                .arg("--target")
+                .arg(target)
+                .arg("--print=cfg")
+                .output()
+                .unwrap()
+                .stdout;
+            let cfgs = String::from_utf8(cfgs).unwrap();
+
+            let expected = target_from_cfgs(target, &cfgs);
+            let actual = Target::from_str(target);
+
+            if Some(&expected) != actual.as_ref().ok() {
+                eprintln!("failed comparing {target}:");
+                eprintln!("  expected: Ok({expected:?})");
+                eprintln!("    actual: {actual:?}");
+                eprintln!();
+                has_failure = true;
+            }
+        }
+
+        if has_failure {
+            panic!("failed comparing targets");
+        }
+    }
+
+    // Various custom target triples not (or no longer) known by `rustc`
+    #[test]
+    fn parse_extra() {
+        let targets = [
+            "arm-frc-linux-gnueabi",
+            "aarch64-uwp-windows-gnu",
+            "arm-unknown-netbsd-eabi",
+            "armv7neon-unknown-linux-gnueabihf",
+            "armv7neon-unknown-linux-musleabihf",
+            "thumbv7-unknown-linux-gnueabihf",
+            "thumbv7-unknown-linux-musleabihf",
+            "x86_64-rumprun-netbsd",
+        ];
+
+        for target in targets {
+            // Check that it parses
+            let _ = Target::from_str(target).unwrap();
+        }
+    }
```

Comment:
These were a few targets I found lying around in the source code. I'm not sure how good `cc`'s support is for targets that require a custom target json file (i.e. most embedded targets)? We should probably extend this table to ensure we also support those.